src/commands/tokens/send_ft/amount_ft.rs

```diff
@@ -113,10 +101,122 @@ impl AmountFt {
             .prompt()?,
         ))
     }
+}
+
+#[derive(Debug, Clone, interactive_clap::InteractiveClap)]
+#[interactive_clap(input_context = AmountFtContext)]
+#[interactive_clap(output_context = FtTransferParamsContext)]
+pub struct FtTransferParams {
+    #[interactive_clap(skip_default_input_arg)]
+    /// Enter a memo for transfer (optional):
+    memo: Option<String>,
+    #[interactive_clap(long = "prepaid-gas")]
+    #[interactive_clap(skip_interactive_input)]
+    gas: Option<crate::common::NearGas>,
+    #[interactive_clap(long = "attached-deposit")]
+    #[interactive_clap(skip_interactive_input)]
+    deposit: Option<crate::types::near_token::NearToken>,
+    #[interactive_clap(named_arg)]
+    /// Select network
+    network_config: crate::network_for_transaction::NetworkForTransactionArgs,
+}
+
+#[derive(Clone)]
+pub struct FtTransferParamsContext(crate::commands::ActionContext);
+
+impl FtTransferParamsContext {
+    pub fn from_previous_context(
+        previous_context: AmountFtContext,
+        scope: &<FtTransferParams as interactive_clap::ToInteractiveClapContextScope>::InteractiveClapContextScope,
+    ) -> color_eyre::eyre::Result<Self> {
+        let get_prepopulated_transaction_after_getting_network_callback: crate::commands::GetPrepopulatedTransactionAfterGettingNetworkCallback =
+            std::sync::Arc::new({
+                let signer_account_id = previous_context.signer_account_id.clone();
+                let ft_contract_account_id = previous_context.ft_contract_account_id.clone();
+                let receiver_account_id = previous_context.receiver_account_id.clone();
+                let ft_transfer_amount = previous_context.ft_transfer_amount.clone();
+                let memo = scope.memo.as_ref().and_then(|s| {
+                    let trimmed = s.trim();
+                    if trimmed.is_empty() {
+                        None
+                    } else {
+                        Some(trimmed.to_string())
+                    }
+                });
```

Comment:
`and_then() + trim` seems to be unnecessary as memo is then also trimmed inside `get_prepopulated_transaction` function.

Reply:
Done